tools/build/libwebrtc/build.ps1

```diff
@@ -0,0 +1,282 @@
+
+# Get the root folder of MixedReality-WebRTC
+function Get-RootFolder
+{
+    $repoRoot = Join-Path $PSScriptRoot "../../../" -Resolve
+    return $repoRoot
+}
+
+# Get the <root>/external/ folder
+function Get-ExternalFolder
+{
+    $repoRoot = Get-RootFolder
+    $externalFolder = Join-Path $repoRoot "external/"
+    return $externalFolder
+}
+
+# Write a message for a task starting, with a specific color to make it
+# stand out between long-running tasks themselves outputting lots of info.
+function Write-TaskStart([string]$message)
+{
+    Write-Host -Foreground Magenta "`n$message"
+}
+
+# Write a failure message without generating any Powershell error.
+function Write-Failure([string]$message)
+{
+    Write-Host -NoNewLine -Foreground Red "Error: "
+    Write-Host $message
+}
+
+# Write a key-value pair with higlight on the value.
+function Write-KeyValue([string]$key, [string]$value)
+{
+    Write-Host -NoNewLine $key
+    Write-Host -Foreground Blue $value
+}
+
+# Initialize the environment variables
+function Initialize-BuildEnvironment
+{
+    Write-TaskStart "Setting up environment..."
+
+    $repoRoot = Get-RootFolder
+    Write-KeyValue "Repository root: " $repoRoot
+
+    # Add depot_tools folder to PATH
+    $externalFolder = Join-Path $repoRoot "external/"
+    $depotToolsPath = Join-Path $externalFolder "depot_tools"
+    if (!($env:PATH.Contains($depotToolsPath)))
+    {
+        # Be sure to append, not prepend. depot_tools/ contains some git version,
```

Comment:
Are we sure that appending is good enough? I am not sure what Google's guidelines say.

Reply:
Yeah [they say "to the front"](https://commondatastorage.googleapis.com/chrome-infra-docs/flat/depot_tools/docs/html/depot_tools_tutorial.html#_setting_up). But the recommendations are for installing `depot_tools` when used with Chromium as I understand. The reason they do that AFAIK is to inject some custom git commands like `git freeze` and `git thaw`, but we don't need those since we don't intend to actually make any change to libwebrtc.

But if I do that then the git version inside `external/depot_tools/` is used and the credential manager _etc._ are not configured, so for example I couldn't push that change anymore to GitHub. So I think it's safe to ignore and use the user's installed git instead, since we don't rely on most tools of `depot_tools`, only `gclient` and `fetch`.

Reply:
Maybe we can prepend and then restore the original path at the end of the script, just to be sure? Or just add a warning that things _shouldn't_ but _might_ break because of this 😄

Reply:
Fixed with scope.